lib/whitehall_importer/integrity_checker.rb

```diff
@@ -87,10 +88,72 @@ def organisation_problems
       problems
     end
 
+    def unpublishing_problems
+      problems = []
+      return problems unless edition.withdrawn? || edition.removed?
+
+      unless expected_state?
+        problems << problem_description(
+          "edition state isn't expected",
+          "unpublished",
+          publishing_api_content.dig("publication_state"),
+        )
+        return problems
+      end
+
+      check = UnpublishingCheck.new(edition, publishing_api_unpublishing)
+
+      unless check.expected_type?
+        problems << unpublishing_type_problem(
+          edition.state,
+          publishing_api_unpublishing["type"],
+        )
+      end
+
+      unless check.expected_alternative_path?
+        problems << problem_description(
+          "unpublishing alternative path doesn't match",
+          publishing_api_unpublishing["alternative_path"],
+          edition.status.details.alternative_url,
+        )
+      end
+
+      unless check.expected_unpublishing_time?
+        problems << problem_description(
+          "unpublishing time doesn't match",
+          publishing_api_unpublishing["unpublished_at"],
+          edition.status.details.withdrawn_at,
```

Comment:
Maybe we want to ask the checker for expected_unpublishing_time it feels a bit opaque why we've used withdrawn_at here

Reply:
I've added `#expected_unpublishing_time` to the Unpublishing Checker.